### DISCUSSION TITLE
I can not get notes with this code

### DISCUSSION THREAD

**Discussion by 2311627BorisBacic** [melanchall/drywetmidi#308] [Q&A] [answered]
sing Microsoft.SqlServer.Server;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Melanchall.DryWetMidi.Core;
using Melanchall.DryWetMidi.Interaction;
using Melanchall.DryWetMidi.MusicTheory;

internal class Program
{
    public static void Main(string[] args)
    {
        MidiFile file = MidiFile.Read("3AkotePitajuSHORT2.mid");
        IEnumerable<Note> notes = file.GetNotes();
        Console.WriteLine("Work");
    }
}

1 Error (active)	CS0104	'Note' is an ambiguous reference between 'Melanchall.DryWetMidi.Interaction.Note' and 'Melanchall.DryWetMidi.MusicTheory.Note'	VrijemeDogadanjaMIDI	C:\Users\boris\source\repos\VrijemeDogadanjaMIDI\VrijemeDogadanjaMIDI\Program.cs	16	

2 Error (active)	CS0104	'Note' is an ambiguous reference between 'Melanchall.DryWetMidi.Interaction.Note' and 'Melanchall.DryWetMidi.MusicTheory.Note'	VrijemeDogadanjaMIDI	

![Slika gdje je zapeo VS u citanju nota](https://github.com/user-attachments/assets/1d37b592-dbaf-43c2-84b3-52fe14a16fda)

**Comment by melanchall** [comment, marked as answer]
Visual Studio has the clear message about what's going on: there is `Note` class in both namespaces - `Melanchall.DryWetMidi.Interaction` and `Melanchall.DryWetMidi.MusicTheory`. If you search the error (_c# is an ambiguous reference between_), you'll find that it's a common situation in .NET world and there are several ways to resolve it.

In your short example you can just write:

```csharp
IEnumerable<Melanchall.DryWetMidi.Interaction.Note> notes = file.GetNotes();
```

or

```csharp
var notes = file.GetNotes();
```

or

```csharp
using InteractionNote = Melanchall.DryWetMidi.Interaction.Note;

...

IEnumerable<InteractionNote> notes = file.GetNotes();
```

**Reply by 2311627BorisBacic** [reply]
Thank s a loot for your support. I am simply new in C# and visual studio world.

**Reply by 2311627BorisBacic** [reply]
I tried this part of code 

var notes = file.GetNotes();
 
working well

but when I put this  

using InteractionNote = Melanchall.DryWetMidi.Interaction.Note;
..
IEnumerable<InteractionNote> notes = file.GetNotes();

and this 

IEnumerable<Melanchall.DryWetMidi.Interaction.Note> notes = file.GetNotes();

In visual studio code terminal return 
can not find file. in file.GetNotes();

**Reply by melanchall** [reply]
@2311627BorisBacic Show please the full content of the file.

**Comment by 2311627BorisBacic** [comment]
using Microsoft.SqlServer.Server;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Melanchall.DryWetMidi.Core;
using Melanchall.DryWetMidi.Interaction;
using Melanchall.DryWetMidi.MusicTheory;
using InteractionNote = Melanchall.DryWetMidi.Interaction.Note;
using System.Security.Cryptography.X509Certificates;
using Microsoft.VisualBasic;

namespace Proba3
{
    class Program
    {
  
        static void Main(string[] args)
       {
          string SONG = "3Ako.mid";
          var midiFile = MidiFile.Read(SONG);
          var notes = midiFile.GetNotes();
        
          // If I remove line 23 above with 
          // version 2 IEnumerable<Melanchall.DryWetMidi.Interaction.Note> notes = file.GetNotes(); file. can not be find or
          // version 3 using InteractionNote = Melanchall.DryWetMidi.Interaction.Note;
          //...
          //IEnumerable<InteractionNote> notes = file.GetNotes(); file. can not be find also

          var TempoMap = midiFile.GetTempoMap();
          
          note mynote = new note();
          
          TempoMap tempoMap = midiFile.GetTempoMap();
          MetricTimeSpan metricTime = mynote.TimeAs<MetricTimeSpan>(tempoMap);
          BarBeatTicksTimeSpan musicalTime = mynote.TimeAs<BarBeatTicksTimeSpan>(tempoMap);

         // public MetricTimeSpan(int hours, int minutes, int seconds, int milliseconds)
         /* long ticks = 123;
            MetricTimeSpan metricTimeSpan = TimeConverter.ConvertTo<MetricTimeSpan>(ticks, tempoMap);
            MetricTimeSpan metricTime2 = metricTimeSpan;
          
          var lyricTimedEvents = midiFile      
                .GetTimedEvents()
                .Where(e=> e.Event.EventType == MidiEventType.NoteOn);   

          int TimeNote = notes.Count;   
          
          

          foreach (var e in lyricTimedEvents) 
          {
            var metricTime = e.TimeAs<MidiTimeSpan>(TempoMap);
            Console.Write($@"
             {TempoMap.TimeDivision} {notes.Count}");
          } */
          
          Console.WriteLine(SONG);

          foreach (var note in midiFile.GetNotes())
          
            {             
             Console.Write($@"
             {note.Time} {note} {notes.Count}");
            }

            
            
            
            Console.WriteLine("        ");
            Console.WriteLine("     ---- MID END");
            
            Console.ReadKey();   
        }
    }
}

**Reply by melanchall** [reply]
Of course the `file` variable cannot be found, because you use `midiFile` variable. This works great:

```csharp
var notes = midiFile.GetNotes();
IEnumerable<Melanchall.DryWetMidi.Interaction.Note> notes2 = midiFile.GetNotes();
IEnumerable<InteractionNote> notes3 = midiFile.GetNotes();
```

Please be more attentive. I'm not a C# teacher :-)

**Reply by 2311627BorisBacic** [reply]
Thank s lot I am started before one month in C# first with visual studio 2022 now I change to Visual Studio Code I am moving but moving very slow. I want to do something with mid file and I found your  GitHub and I started.
